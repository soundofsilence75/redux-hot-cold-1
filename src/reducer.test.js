import { reducer } from "./reducer";
import { restartGame, makeGuess } from "./actions";

describe("reducer", () => {
  const guesses = [1, 56, 99];
  const feedback = "Make your guess!";
  const correctAnswer = 50;
  const auralStatus = "";

  it("Should return the current state on an unknown action", () => {
   let currentState = {};
   const state = reducer(currentState, {type: "__UNKNOWN"});
   expect(state).toBe(currentState);
  });

  describe("restartGame", () => {
    it("Should restart the game", () => {
      let state;
      state = reducer(state, restartGame());
      expect(state).toEqual({
        auralStatus,
        correctAnswer: undefined,
        feedback,
        guesses: []
      });
    });
  });

  describe("makeGuess", () => {
    it("Should make a new guess", () => {
      let state;
      let guess = 42;
      state = reducer(state, makeGuess(guess));
      const { guesses } = reducer(state, makeGuess(guess));
      expect ({ guesses }).toMatchSnapshot();
    });
  });
});