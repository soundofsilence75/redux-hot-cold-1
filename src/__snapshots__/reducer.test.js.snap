// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer makeGuess Should make a new guess 1`] = `
Object {
  "guesses": Array [
    42,
    42,
  ],
}
`;
